File: app/javascript/channels/temperature_channel.js
import consumer from "channels/consumer"

consumer.subscriptions.create("TemperatureChannel", {
  connected() {
    console.log("Connected to TemperatureChannel");
  },

  disconnected() {
    // Called when the subscription has been terminated by the server
  },

  received(data) {
    if (data.action === "analog") {
      console.log(data);

      const intValue = parseInt(data.value, 10);
      const targetElement = document.getElementById(`temp_${data.index}`);
      if (targetElement) { // 요소가 존재하는지 확인
        targetElement.innerText = intValue;
      }
    }
  }
});

File: app/channels/temperature_channel.rb
class TemperatureChannel < ApplicationCable::Channel
  def subscribed
    stream_from "temperature_channel"
  end

  def unsubscribed
    puts "##### disconnected"
    # Any cleanup needed when channel is unsubscribed
  end

  def analog(data)
    Rails.logger.info "Received data on TemperatureChannel: #{data.inspect}"
    ActionCable.server.broadcast("temperature_channel",data)
  end

  def digital(data)
    Rails.logger.info "Received data on TemperatureChannel: #{data.inspect}"
    ActionCable.server.broadcast("temperature_channel",data)
  end

end
